packages/gasket-plugin-nextjs/lib/setup-next-app.js

```diff
@@ -9,13 +9,19 @@ const { createConfig } = require('./config');
  * @private
  */
 async function setupNextApp(gasket) {
-  const { exec, command } = gasket;
+  const { exec, command, config } = gasket;
+  const { hostname, http, https, http2 } = config;
   const createNextApp = require('next');
   const devServer = (command.id || command) === 'local';
 
+  const _http = http || https || http2;
+  const port = _http.port || _http;
+
   const app = createNextApp({
     dev: devServer,
-    conf: await createConfig(gasket, devServer)
+    conf: await createConfig(gasket, devServer),
+    hostname,
```

Comment:
I don't think `hostname` here is the same as what the next server wants; hostname in the case of Gasket is the end-user's hostname, whereas the next server uses it for the IP address it'll bind the listener to. The gasket config should use the [`*.host`](https://github.com/http-party/create-servers#usage) setting for that.

Of course if we're never going to call `.listen(...)` on this it probably doesn't matter.

Reply:
I may be slightly confused by your comment. I opened [this](https://github.com/vercel/next.js/issues/39666) issue with the Next.js team thinking that it was a bug. The solution they outlined requires that we pass the hostname and port when creating the next app. After implementing this fix and testing with the Gasket Canary App, the app functioned as expected.

Reply:
It sounds like the constructor needs `hostname` because of its validation, but I doubt they actually use it unless you call `.listen()`. If you _are_ planning on using their `listen()` call then you'll need to use the `host` listener properties (the address that's being listened on), not `hostname` which is something different.